File: front_end/lib/services/stage_log.dart
import 'package:flutter/foundation.dart';

/// Clear stage-by-stage logs in the Flutter terminal (`flutter run`).
class StageLog {
  StageLog._();

  static void step(String stage, String message, [Map<String, Object?>? extra]) {
    final buf = StringBuffer('[ConvoBridge][$stage] $message');
    if (extra != null && extra.isNotEmpty) {
      final detail = extra.entries.map((e) => '${e.key}=${e.value}').join(' | ');
      buf.write(' ($detail)');
    }
    debugPrint(buf.toString());
  }

  static void boot(String aiBaseUrl) {
    step('BOOT', 'App starting', {'aiServer': aiBaseUrl});
  }
}

File: app/core/stage_log.py
"""Clear stage-by-stage terminal logs for ConvoBridge backend demos."""

from __future__ import annotations

from datetime import datetime


def stage(step: str, message: str, **extra: object) -> None:
    """Print one human-readable pipeline stage line to the server terminal."""
    ts = datetime.now().strftime("%H:%M:%S")
    bits = [f"[{ts}]", "[ConvoBridge]", f"[{step}]", message]
    if extra:
        detail = " | ".join(f"{k}={v}" for k, v in extra.items())
        bits.append(f"({detail})")
    print(" ".join(bits), flush=True)


def banner() -> None:
    stage(
        "BOOT",
        "AI backend ready — watching for /health /stt /translate /summarize /chatbot",
    )
